src/coreComponents/linearAlgebra/interfaces/hypre/HypreVector.hpp

```diff
@@ -244,29 +191,25 @@ class HypreVector final : private VectorBase< HypreVector >
 
   ///@}
 
+private:
+
+  /// ParVector pointer alias
+  using HYPRE_ParVector = hypre_ParVector *;
+
+public:
+
   /**
    * @brief Returns a pointer to the implementation.
    * @return the underlying HYPRE_ParVector object.
    */
   HYPRE_ParVector const & unwrapped() const;
 
-  /**
-   * @brief Returns a pointer to the implementation.
-   * @return the underlying HYPRE_IJVector object.
-   */
-  HYPRE_IJVector const & unwrappedIJ() const;
-
 private:
 
-  /**
-   * Pointer to underlying HYPRE_IJVector type.
-   */
-  HYPRE_IJVector m_ij_vector;
-
   /**
    * Pointer to underlying HYPRE_ParVector type.
    */
-  HYPRE_ParVector m_par_vector;
+  HYPRE_ParVector m_vec;
```

Comment:
Is there a specific reason to keep using `HYPRE_ParVector`? Given the ongoing refactoring, would it make sense to use directly `hypre_ParVector`?

Reply:
No particular technical reason, but in theory clients of `HypreVector` class might not be including "private" hypre headers, and the only definitions they would have access to from hypre are these:
https://github.com/hypre-space/hypre/blob/66e1f2df45331ddc01a424586ad86716d8d75367/src/parcsr_mv/HYPRE_parcsr_mv.h#L30-L31
```c
struct hypre_ParVector_struct;
typedef struct hypre_ParVector_struct *HYPRE_ParVector;
```
`HYPRE_ParVector` is also the name of the type in most other hypre function signatures (e.g. all BoomerAMG/Krylov calls) so I would want `unwrapped()` to return exactly that type to avoid confusion (even though it's just an alias for `hypre_ParVector *`). This is also how we do things in `PetscVector` where `Vec` is the "opaque handle" type stored and returned, not `_p_Vec *`.